pkg/integrations/v2/mssql/sql_exporter.go

```diff
@@ -0,0 +1,149 @@
+type Config struct {
+	ConnectionString   string               `yaml:"connection_string,omitempty"`
+	MaxIdleConnections int                  `yaml:"max_idle_connections,omitempty"`
+	MaxOpenConnections int                  `yaml:"max_open_connections,omitempty"`
+	Timeout            time.Duration        `yaml:"timeout,omitempty"`
+	Common             common.MetricsConfig `yaml:",inline"`
+}
+
+func (c Config) validate() error {
+	if c.ConnectionString == "" {
+		return errors.New("the connection_string parameter is required")
+	}
+
+	url, err := url.Parse(c.ConnectionString)
+	if err != nil {
+		return fmt.Errorf("failed to parse connection_string: %w", err)
+	}
+
+	if url.Scheme != "sqlserver" {
+		return errors.New("scheme of provided connection_string URL must be sqlserver")
+	}
+
+	if c.MaxOpenConnections < 1 {
+		return errors.New("max_connections must be at least 1")
+	}
+
+	if c.MaxIdleConnections < 1 {
+		return errors.New("max_idle_connection must be at least 1")
+	}
+
+	if c.Timeout <= 0 {
+		return errors.New("timeout must be positive")
+	}
+
+	return nil
+}
+
+// ApplyDefaults applies the integration's default configuration.
+func (c *Config) ApplyDefaults(globals integrations_v2.Globals) error {
+	c.Common.ApplyDefaults(globals.SubsystemOpts.Metrics.Autoscrape)
+	return nil
+}
+
+// Identifier returns a string that identifies the integration.
+func (c *Config) Identifier(globals integrations_v2.Globals) (string, error) {
+	if c.Common.InstanceKey != nil {
+		return *c.Common.InstanceKey, nil
+	}
+
+	url, err := url.Parse(c.ConnectionString)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse connection string URL: %w", err)
+	}
+
+	return url.Host, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Name returns the name of the integration this config is for.
+func (c *Config) Name() string {
+	return "mssql"
+}
+
+func init() {
+	integrations_v2.Register(&Config{}, integrations_v2.TypeMultiplex)
+}
+
+// NewIntegration creates a new v2 integration from the config.
+func (c *Config) NewIntegration(l log.Logger, globals integrations_v2.Globals) (integrations_v2.Integration, error) {
+	err := c.validate()
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate config: %w", err)
+	}
+
+	handler, err := createHandler(l, c)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create http handler: %w", err)
+	}
+
+	return metricsutils.NewMetricsHandlerIntegration(l, c, c.Common, globals, handler)
+}
+
+func createHandler(logger log.Logger, c *Config) (http.HandlerFunc, error) {
+	t, err := sql_exporter.NewTarget(
+		"mssqlintegration",
+		"",
+		c.ConnectionString,
+		[]*config.CollectorConfig{
+			&mssql_common.CollectorConfig,
+		},
+		prometheus.Labels{},
+		// TODO: Evaluate if these need to be configurable
```

Comment:
Which value? `TimeoutOffset`?

Reply:
Removed this comment; Was leftover before I added options for max connections and scrape timeout added.